analysis/quantifier-comparisons/14-cellhash-exploration.Rmd

```diff
@@ -0,0 +1,371 @@
+```{r}
+ggplot(hashdrops_classification, aes(x = hashdrops_logfc)) +
+  geom_histogram(bins = 50)+ 
+  labs(x = "Log Fold Change from first to second HTO")
+```
+Not much with even a 2-fold change, which makes it very hard to assign cells confidently.
+We could lower the threshold, but that seems fraught.
+
+
+### Calculating a separate ambient set
+
+We may be able to use the unfiltered data to assign a better ambient profile. Let's explore that:
+
+First we will pull out the "empty" cells
+```{r}
+empty_cells <- ! colnames(hash_sce_unfiltered) %in% colnames(hash_sce)
+hash_sce_empty <- hash_sce_unfiltered[,empty_cells]
+```
+Now we will sum all cells to calculate the ambient profile.
+There are many cells here with no calls at all, but they will not contribute much at all to the profile so we will ignore those.
+```{r}
+ambient_sum <- rowSums(counts(hash_sce_empty))
+ambient <- ambient_sum / sum(ambient_sum)
+ambient
+```
+
+Now we can try recalculating hash results with this ambient profile:
+```{r}
+hash_result2 <- DropletUtils::hashedDrops(hash_sce, ambient = ambient)
+
+sum(hash_result2$Confident)
+```
+Are we getting the same cells called confidently?
+```{r}
+sum(hash_result$Confident & hash_result2$Confident)
+```
+
+
+## Assigning cells with Seurat
+
+```{r}
+# create a Seurat object
+seurat_obj <- Seurat::CreateSeuratObject(counts(sce_filtered))
+seurat_obj[["HTO"]] <- CreateAssayObject(counts = counts(hash_sce))
+seurat_obj <- Seurat::NormalizeData(seurat_obj, assay = "HTO", normalization.method = "CLR")
+seurat_obj <- Seurat::HTODemux(seurat_obj)
+```
+
+This worked, though with at least one other sample it failed for unknown reasons, so we may not want to rely on it.
+
+I'm also generally uncomfortable with the fact that it seems to call samples based on a relatively simple cutoff for each HTO; 
+
+Let's look at the results:
+
+```{r}
+table(seurat_obj$HTO_classification.global)
+```
+So it looks like it did make calls for a much larger number of cells, but the majority are still not being assigned to a sample, and there are a very large number of "doublets."
+How confident are those calls?
+
+```{r}
+# Group cells based on the max HTO signal
+Idents(seurat_obj) <- "HTO_maxID"
+RidgePlot(seurat_obj, assay = "HTO", features = rownames(seurat_obj[["HTO"]])[1:4], ncol = 2)
+```
+
+I would not call those super-impressive differences.
+A very slight upward bias for th matching oligo, but that would be expected given the procedure being used to assign the cells requires it.
+
+Overall, I am not confident that the called cells are very reliable based on these data. 
+
+
+Let's pull out the identities for comparison later.
+```{r}
+seurat_classification <- data.frame(
+  cell_barcode = colnames(seurat_obj),
+  seurat_id = as.character(seurat_obj$hash.ID),
+  seurat_maxid = as.character(seurat_obj$HTO_maxID),
+  seurat_margin = seurat_obj$HTO_margin
+)
```

Comment:
Similar to my previous comment about overlaying the HTO classification on top of the PCA, I think you could do the same thing here. I know you do it below with the expression data, but there you see that the clustering is based on RNA expression and cells for each sample are intermixed within each cluster. Theoretically, if you have good separation of cells classified by hashing oligos, they should separate based on HTO expression (which you don't really see in the ridge plot). To be able to compare how Seurat is performing in comparison the `emptyDrops` I think it would be helpful to also plot the Seurat classification on the PCA calculated using HTO expression and see how they separate.